dirty_cat/test/test_minhash_encoder.py

```diff
@@ -56,25 +58,76 @@ def test_get_unique_ngrams():
     enc = MinHashEncoder(n_components=2)
     ngrams = enc.get_unique_ngrams(string, ngram_range)
     assert ngrams == true_ngrams
-        
+
+
 def profile_encoder(Encoder, hashing='fast', minmax_hash=False):
     # not an unit test
-    
+
     from dirty_cat import datasets
     import pandas as pd
     employee_salaries = datasets.fetch_employee_salaries()
-    data = pd.read_csv(employee_salaries['path'])
-    X = data['Employee Position Title'].tolist()
+    data = employee_salaries['data']
+    X = data['employee_position_title'].tolist()
     X = X * 10
     t0 = time.time()
-    encoder = Encoder(n_components=50, hashing=hashing, minmax_hash=minmax_hash)
+    encoder = Encoder(n_components=50, hashing=hashing,
+                      minmax_hash=minmax_hash)
     encoder.fit(X)
     y = encoder.transform(X)
     assert y.shape == (len(X), 50)
     eta = time.time() - t0
     return eta
 
 
+def _test_missing_values(input_type, missing):
+    X = ['Red',
+         np.nan,
+         'green',
+         'blue',
+         'green',
+         'green',
+         'blue',
+         float('nan')]
+    n = 3
+    z = np.zeros(n)
+
+    if input_type == 'numpy':
+        X = np.array(X, dtype=object)
+    elif input_type == 'pandas':
+        pd = pytest.importorskip("pandas")
+        X = pd.DataFrame(X)
+
+    for hashing in ['fast', 'murmur']:
+        encoder = MinHashEncoder(n_components=n, hashing=hashing,
+                                 minmax_hash=False, handle_missing=missing)
+        if missing == 'error':
+            encoder.fit(X)
+            if input_type in ['numpy', 'pandas']:
+                msg = ("Found missing values in input data; set "
+                       "handle_missing='' to encode with missing values")
+                assert_raise_message(ValueError, msg, encoder.transform, X)
+        elif missing == '':
+            encoder.fit(X)
+            y = encoder.transform(X)
+            if input_type == 'list':
+                assert np.allclose(y[1], y[-1])
+            else:
+                assert np.array_equal(y[1], z)
+                assert np.array_equal(y[-1], z)
+        else:
+            msg = "handle_missing should be either 'error' or '', got %s" % missing
+            assert_raise_message(ValueError, msg, encoder.fit_transform, X)
+            return
+
+
+def test_missing_values():
```

Comment:
Maybe here we should be using pytest.parametrize:

https://docs.pytest.org/en/stable/parametrize.html

Reply:
fixed